content/academy/anti_scraping/techniques/rate_limiting.md

```diff
@@ -16,9 +16,40 @@ In cases when a higher number of requests is expected for the crawler, using a [
 
 ## [](#handling-rate-limiting) Dealing with sites which have rate-limiting
 
-In the Apify SDK, you can use [`browserPoolOptions.retireBrowserAfterPageCount`](https://github.com/apify/browser-pool#features) for browser-based crawlers (Puppeteer/Playwright) to retire a browser instance after a certain number of requests have been sent from it. In request-based crawlers, the [Session Pool](https://sdk.apify.com/docs/api/session-pool) can be leveraged.
-
-<!-- Add more here -->
+In the Apify SDK, you can use [`browserPoolOptions.retireBrowserAfterPageCount`](https://github.com/apify/browser-pool#features) for browser-based crawlers (Puppeteer/Playwright) to retire a browser instance after a certain number of requests have been sent from it. This will prevent the browser instance from sending too many requests to the page, avoiding the potential of hitting rate-limiting issues:
+
+```JavaScript
+import Apify from 'apify';
+
+const myCrawler = new Apify.PuppeteerCrawler({
+    browserPoolOptions: {
+        // Let's say the website starts blocking requests after
+        // 20 requests have been sent in the span of 1 minute from
+        // a single user.
+        // We can stay on the safe side and retire the browser
+        // after 15 pages (requests) have been opened.
+        retireBrowserAfterPageCount: 15,
+    },
+    // ...
+});
+```
+
+In request-based crawlers, the [Session Pool](https://sdk.apify.com/docs/api/session-pool) or a similar implementation of it can be leveraged:
```

Comment:
It works with browsers as well.

I think this whole section could be remade by focusing on session pool and how to set it up properly for different scenarios. You can find examples of it in Web Scraper. You'll need to study the code a bit to understand how it works. I'm happy to help you understand the whole concept, if needed.

https://github.com/apify/actor-scraper/blob/2d862c4eb4766a4f3c5f575713e2f387ac9fd422/web-scraper/src/crawler_setup.js#L229-L244